File: 167.two-sum-ii-input-array-is-sorted.js
/*
 * @lc app=leetcode id=167 lang=javascript
 *
 * [167] Two Sum II - Input array is sorted
 */
/**
 * @param {number[]} numbers
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function(numbers, target) {
	//不是追击类的，相遇类的
	var i = 0,
		j = numbers.length - 1;
	while (i < j) {
		if (target == numbers[i] + numbers[j]) return [ i + 1, j + 1 ];
		else if (target > numbers[i] + numbers[j]) i++;
		else j--;
	}
	return [];
};

File: java/167.two-sum-ii-input-array-is-sorted.java
/*
 * @lc app=leetcode id=167 lang=java
 *
 * [167] Two Sum II - Input array is sorted
 */

// @lc code=start
class Solution {
    public int[] twoSum(int[] nums, int target) {
        int i = 0, j = nums.length - 1;
        // i == j是没有意义的
        while (i < j) {
            if (nums[i] + nums[j] == target)
                return new int[] { i + 1, j + 1 };
            else if (nums[i] + nums[j] > target) {
                j--;
            } else {
                i++;
            }
        }
        throw new Error("input no solution");
    }
}
// @lc code=end
